### DISCUSSION TITLE
Importing and updating only a sub-region

### DISCUSSION THREAD

**Discussion by navigatrum** [osm-search/Nominatim#2520] [Q&A] [answered]
What is the proper way to import only one administrative sub-region contained in a OSM extract file?

I'm performing this task with Osmconvert:
1) downloading the smallest super-region of my adminitrative sub-region from Geofabrik
2) downloading the `.poly` file of my administrative sub-region from https://polygons.openstreetmap.fr
3) clipping my sub-region with osmconvert: 
```osmconvert super-region.osm.pbf -B="sub-region.poly" -o=sub-region.osm.pbf```
4) passing `sub-region.osm.pbf` to Nominatim.

Which works fine, but with a such workflow I can't use Nominatim's auto-update features.

So I'm wondering if there is an alternative way to perform such a common operation.

Or should I necessarly implement an `Unit service` running periodically a shell script  to do the osmconvert stuff? And in that case, is there any way to use `pyosmium-get-changes` and `nominatim add-data --diff` (as is done for multiple region maps in the utils script [update_database.sh](https://github.com/osm-search/Nominatim/blob/master/utils/update_database.sh)) to just reindex the database with the updates relative to my sub-region? Or does my sub-region database need to be re-created from scratch every time changes occurr in the `super-region.osc.gz`?

[edit: corrected url]

**Comment by mtmail** [comment, marked as answer]
There's no tutorial or script yet that deals with non Geofabrik extracts.

The `osmium` tool is a bit more powerful and you wouldn't have to deal with `.poly` files. https://osmcode.org/osmium-tool/manual.html#creating-geographic-extracts 

If you version your `.osm.pbf` files, e.g. by date, then you can create the change files files `.osc.gz` with `osmium derive-changes` and call `pyosmium-get-changes` like you see in `update_database.sh`.

I'm not sure how comm

**Reply by navigatrum** [reply]
Thanks! So `osmium derive-changes` seems the way to go.

And just to dispel any doubts, may I ask you if you can confirm that there is no way to set an [import style](https://nominatim.org/release-docs/develop/customize/Import-Styles/) in such a way that only what is related to ~~a given city, or province, which is~~ one of the cities which are 'contained' in a given Geofabrik extract, will be imported and updated into the db?

**Reply by mtmail** [reply]
You're right. The import style only select the tags (the 'what'), not the bounds (the 'where') during the import.

**Comment by lonvia** [comment]
openstreetmap.fr has extracts with matching minutely updates. If you use their polygons, then you likely can use those extracts directly. The extracts are at  https://download.openstreetmap.fr/extracts/ and the replication files can be found below https://download.openstreetmap.fr/replication/ They should work nicely together with Nominatim's update mechanism.

**Reply by navigatrum** [reply]
Nice! The subregion I have to clip with Osmconvert starting from a Geofabrik extract it's already clipped in openstreetmap.fr.
